### ISSUE TITLE
Define all requirements required in setup.py file on the extension. 

### ISSUE THREAD

**Issue by sujen1412** [MAAP-Project/jupyter-server-extension#4]
The current one requires the following

```
jupyter_packaging==0.12.3
jupyter_server==2.13.0
maap==0.2.1
nbformat==5.10.3
notebook==7.1.2
pytest==8.1.1
PyYAML==6.0.1
Requests==2.31.0
setuptools==68.2.2
tornado==6.4
xmltodict==0.13.0
```

**Comment by grallewellyn** [comment]
BTW going use jupyter_server==2.12.5 because of this PR fixing a bug: https://github.com/MAAP-Project/jupyter-server-extension/pull/2/files

**Comment by sujen1412** [comment]
Feel free to change the versions as needed. This is just a template.

**Comment by grallewellyn** [comment]
@sujen1412 why would I need to add git+https://github.com/MAAP-Project/maap-py.git@v3.1.4#egg=maapPy to the install_requires?
Both are installed when we build our images so this would just be another place we need to change the version of maap-py

**Comment by sujen1412** [comment]
Agreed that it is installed when building the image. But how would we know if updating maap-py in the image will break this extension or not ? 

This extension uses maap-py in its code here - https://github.com/MAAP-Project/jupyter-server-extension/blob/main/jupyter_server_extension/handlers.py#L9. 

So if we were to say build a base image without maap-py in it, this extension would fail as it is relying on something else to provide its dependency (which also was the root cause of the xmltodict error we are seeing now). 

So yes this would be another place we would have to change the version, but in my opinion that is how it should be, independent. 
Once we move to pypi, we could change the setup.py to say install `maap-py>=3.0,<4.0`
